### PR TITLE
Add helper function to raise first encountered exception in `TasksSummary`

### PR DESCRIPTION
**Pull request by pratuat**
## Description

Add helper function to raise first encountered exception in `utils._concurrency.TaskSummary`.

Issue: https://cognitedata.atlassian.net/browse/DOGE-44

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by haakonvt — CHANGES_REQUESTED** [review]
Good change! This will let us more easily fix the issue of not using more narrow exceptions in the next major version bump! 👌

**Comment by haakonvt** [comment]
I found a few more:

image

File: cognite/client/_api/relationships.py [hunk 1]
```diff
@@ -311,8 +311,9 @@ def list(
                 tasks,
                 max_workers=self._config.max_workers,
             )
-            if tasks_summary.exceptions:
-                raise tasks_summary.exceptions[0]
+
+            tasks_summary.raise_first_encountered_exception()
+
```
Comment (haakonvt) [inline cognite/client/_api/relationships.py:316]: Consider if adding space above and below is necessary for readability.

**Comment by pratuat** [comment]
> I found a few more:
> 
> image

sorry I missed those and thank you for finding them 👍 :)

**Review by haakonvt — APPROVED** [review]
Lgtm!

**Pull request merged**